python-sdk/nuscenes/eval/detection/tests/test_algo.py

```diff
@@ -257,5 +281,131 @@ def test_two_imgs(self):
+
+        # One GT and one detection
+        self.check_tp({'sample1': [gt2]}, {'sample1': [pred2]}, target_error=1, metric_name='trans_err')
+
+        # Two GT's and two detections.
+        # The target is the average value of the recall vs. Error curve.
+        target_error = ((0 + 0) / 2 + (0 + 0.5) / 2) / (2 * 0.9)  # It is a piecewise linear with 2 segments.
+        self.check_tp({'sample1': [gt1, gt2]}, {'sample1': [pred1, pred2]}, target_error=target_error,
+                      metric_name='trans_err')
+
+        # Adding a false positive with smaller detection score should not affect the true positive metric.
+        self.check_tp({'sample1': [gt1, gt2]}, {'sample1': [pred1, pred2, pred3]}, target_error=target_error,
+                      metric_name='trans_err')
+
+        target_error = ((0+0)/2 + (0+0.5)/2 + (0.5 + 0.33)/2) / (3 * 0.9)  # It is a piecewise linear with 3 segments
+        self.check_tp({'sample1': [gt1, gt2, gt3]}, {'sample1': [pred1, pred2, pred4]}, target_error=target_error,
+                      metric_name='trans_err')
+
+        # All the detections does have same error, so the overall error is also same.
+        self.check_tp({'sample1': [gt2, gt3]}, {'sample1': [pred2, pred5]}, target_error=1.0,
+                      metric_name='trans_err')
+
+    def test_two_imgs(self):
+        """ Test the more than one sample case. """
+
+        # Note all the following unit tests can be repeated to other metrics, but they are not needed.
+        # The intention of these tests is to measure the calc_tp function which is common for all metrics.
+
+        gt1 = {'trans': (1, 1, 1)}
+        gt2 = {'trans': (10, 10, 1), 'size': (2, 2, 2)}
+        gt3 = {'trans': (20, 20, 1), 'size': (2, 4, 2)}
+
+        pred1 = {'trans': (1, 1, 1), 'score': 1.0}
+        pred2 = {'trans': (11, 10, 1), 'size': (2, 2, 2), 'score': 0.9}
+        pred3 = {'trans': (100, 10, 1), 'size': (2, 2, 2), 'score': 0.8}
+        pred4 = {'trans': (21, 20, 1), 'size': (2, 4, 2), 'score': 0.7}
+
+        # One GT and one detection
+        self.check_tp({'sample1': [gt2]}, {'sample1': [pred2]}, target_error=1, metric_name='trans_err')
+
+        # Two GT's and two detections.
+        # The target is the average value of the recall vs. Error curve.
+        target_error = ((0 + 0) / 2 + (0 + 0.5) / 2) / (2 * 0.9)  # It is a piecewise linear with 2 segments.
+        self.check_tp({'sample1': [gt1], 'sample2': [gt2]}, {'sample1': [pred1], 'sample2': [pred2]},
+                      target_error=target_error, metric_name='trans_err')
+
+        # Adding a false positive and/or an empty sample should not affect the score
+        self.check_tp({'sample1': [gt1], 'sample2': [gt2], 'sample3': []},
+                      {'sample1': [pred1], 'sample2': [pred2, pred3], 'sample3': []},
+                      target_error=target_error, metric_name='trans_err')
+
+        # All the detections does have same error, so the overall error is also same.
```

Comment:
Both matches have same translational error (1 meter), so the overall error is also 1 meter